### PR TITLE
B2B auto login

### PR DESCRIPTION
**Pull request by paddyo**
## Description

After a successful signup, the user is logged into the ft.com and Next.

### Depends on:
https://github.com/Financial-Times/next-signup/pull/655 [ Tests will fail until merge ]

### CODE REVIEW

File: client/components/login-pixel/main.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+export default class LoginPixel {
+
+	constructor({selector='body'} = {}) {
+		this.selector = selector;
+	}
+
+	insert() {
+		let loginPixel = document.createElement('img');
```
Comment (ironsidevsquincy) [inline client/components/login-pixel/main.js]: `const`

File: server/apis/login.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+			email,
+			password,
+			rememberMe: persist
+		};
+
+		let options = {
+			method: 'POST',
+			headers: getHeaders(useTestApi),
+			body: JSON.stringify(payload)
+		};
```
Comment (ironsidevsquincy) [inline server/apis/login.js]: These can be consts, right?

File: server/apis/login.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+		return fetch(`https://${host}/login`, options)
+			.then(handle)
+			.then(({ sessionToken, secureSessionToken }) => ({ sessionToken, secureSessionToken }));
```
Comment (ironsidevsquincy) [inline server/apis/login.js]: `catch`,  or are you depending on the thing calling it to catch?

File: server/routes/api/b2b/trial/post.js [hunk 4]
```diff
@@ -34,12 +37,28 @@ export default function(req, res, next) {
 				return RTS.redeem(token, response.user.id, { useTestApi });
 			})
 			.then(() => {
+				return Login.auth(user.email, user.password, { persist: true, useTestApi });
+			})
```
Comment (ironsidevsquincy) [inline server/routes/api/b2b/trial/post.js]: could concise this

File: server/routes/api/b2b/trial/post.js [hunk 5]
```diff
@@ -34,12 +37,28 @@ export default function(req, res, next) {
...
 	})
 	.catch(next);
 
+	function setSessionCookie(auth) {
+		let expires = new Date(Date.now() + SESSION_COOKIE_EXPIRY);
+		let cookie = {
+			path: '/',
+			domain: '.ft.com',
+			expires
+		}
```
Comment (ironsidevsquincy) [inline server/routes/api/b2b/trial/post.js]: more consts

File: server/routes/api/b2b/trial/post.js [hunk 6]
```diff
@@ -2,13 +2,16 @@ import Crypto from '../../../../libs/crypto';
...
 import RTS from '../../../../apis/redeemable-token-service';
+import Login from '../../../../apis/login';
 
 import MaskLogger from '@financial-times/n-mask-logger';
 
 const logger = new MaskLogger(['email', 'password']);
 
 export default function(req, res, next) {
 
+	const SESSION_COOKIE_EXPIRY = 15552000000; // 6 Months
```
Comment (ironsidevsquincy) [inline server/routes/api/b2b/trial/post.js]: maybe just for ease of changing, `60 * 60 * 24 * 30 * 6`

File: server/apis/login.js (hunk 3 above)
Comment (paddyo) [inline server/apis/login.js]: Yes, caller must catch.

File: server/apis/login.js (hunk 2 above)
Comment (paddyo) [inline server/apis/login.js]: Can be, but it's ambiguous because it's not a deep freeze ... really on the fence about this!

Comment (ironsidevsquincy) [inline server/apis/login.js]: As long as `options` itself isn't being reassigned, it should be fine

File: client/components/login-pixel/main.js (hunk 1 above)
Comment (paddyo) [inline client/components/login-pixel/main.js]: fixed in #659

File: server/apis/login.js (hunk 2 above)
Comment (paddyo) [inline server/apis/login.js]: fixed in #659

File: server/routes/api/b2b/trial/post.js (hunk 4 above)
Comment (paddyo) [inline server/routes/api/b2b/trial/post.js]: fixed in #659

File: server/routes/api/b2b/trial/post.js (hunk 5 above)
Comment (paddyo) [inline server/routes/api/b2b/trial/post.js]: fixed in #659

**Pull request merged**